updatehub/tests/common.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright (C) 2020 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use mockito::{mock, Mock};
+use regex::Regex;
+use serde_json::json;
+use std::{env, path::PathBuf};
+
+pub enum FakeServer {
+    NoUpdate,
+    HasUpdate(String),
+}
+
+pub enum Polling {
+    Enable,
+    Disable,
+}
+
+pub enum Server {
+    Custom(String),
+    Standard,
+}
+
+pub struct Settings {
+    pub polling: bool,
+    pub listen_socket: String,
+    pub server_address: String,
+    pub download_dir: Option<PathBuf>,
+    pub config_file: Option<PathBuf>,
+}
+
+impl Default for Settings {
+    fn default() -> Self {
+        Settings {
+            polling: false,
+            listen_socket: String::from("localhost:8080"),
+            server_address: mockito::server_url(),
+            download_dir: None,
+            config_file: None,
+        }
+    }
+}
+
+impl Settings {
+    pub fn init_server(self) -> (rexpect::session::PtySession, updatehub::tests::TestEnvironment) {
+        let setup = updatehub::tests::TestEnvironment::build()
+            .listen_socket(self.listen_socket)
+            .server_address(self.server_address)
+            .add_echo_binary("reboot");
+
+        let mut setup = if !self.polling { setup.disable_polling() } else { setup }.finish();
+
+        if let Some(download_dir) = self.download_dir {
+            setup.settings.data.update.download_dir = download_dir;
+
+            let content =
+                toml::ser::to_string_pretty(&setup.settings.data.update.download_dir).unwrap();
+            std::fs::write(&setup.settings.stored_path, content).unwrap();
+        }
+
+        if let Some(config_file) = self.config_file {
+            setup.settings.stored_path = config_file;
+        }
+
+        println!("{:#?}", setup.settings.data);
+
+        let cmd = format!(
+            "{} server -v trace -c {}",
+            cargo_bin("updatehub").to_string_lossy(),
+            setup.settings.stored_path.to_string_lossy()
+        );
+
+        let handle = rexpect::spawn(&cmd, None).expect("fail to spawn server command");
```

Comment:
Making the server run with a predefined timeout would be better for development so we can avoid zombie processes.